### PR TITLE
feat(iceberg): support create iceberg table w/ one layer nested type & scan table w/ nested type

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Support nested type with one layer, i.e. the internal field of nested type is a primitive type.
- Resolve https://github.com/risingwavelabs/risingwave/issues/18363#issuecomment-2340432750

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -143,9 +143,10 @@ arrow-array-iceberg = { package = "arrow-array", version = "52" }
 arrow-schema-iceberg = { package = "arrow-schema", version = "52" }
 arrow-buffer-iceberg = { package = "arrow-buffer", version = "52" }
 arrow-cast-iceberg = { package = "arrow-cast", version = "52" }
-iceberg = "0.3.0"
-iceberg-catalog-rest = "0.3.0"
-iceberg-catalog-glue = "0.3.0"
+# branch dylan/merge_on_read
+iceberg = { git = "https://github.com/risingwavelabs/iceberg-rust.git", rev = "df076a9949e9ed6431a6dd3998ac0c49152dda9c" }
+iceberg-catalog-rest = { git = "https://github.com/risingwavelabs/iceberg-rust.git", rev = "df076a9949e9ed6431a6dd3998ac0c49152dda9c" }
+iceberg-catalog-glue = { git = "https://github.com/risingwavelabs/iceberg-rust.git", rev = "df076a9949e9ed6431a6dd3998ac0c49152dda9c" }
```
Comment (chenzl25) [inline Cargo.toml:149]: This branch removes the restriction of nested type for scan.

File: src/common/src/array/arrow/arrow_iceberg.rs [hunk 2]
```diff
@@ -171,6 +173,90 @@ impl ToArrow for IcebergArrowConvert {
...
+            DataType::Serial => self.serial_type_to_arrow(),
+            DataType::Decimal => return Ok(self.decimal_type_to_arrow(name)),
+            DataType::Jsonb => return Ok(self.jsonb_type_to_arrow(name)),
+            DataType::Struct(fields) => self.struct_type_to_arrow(fields)?,
+            DataType::List(datatype) => self.list_type_to_arrow(datatype)?,
+            DataType::Map(datatype) => self.map_type_to_arrow(datatype)?,
+        };
+
+        let mut arrow_field = arrow_schema::Field::new(name, data_type, true);
+        self.add_field_id(&mut arrow_field);
```
Comment (xxchan) [inline src/common/src/array/arrow/arrow_iceberg.rs:255]: `self` in the trait method signature finally becomes useful 🤣

File: src/common/src/array/arrow/arrow_iceberg.rs [hunk 3]
```diff
@@ -171,6 +173,90 @@ impl ToArrow for IcebergArrowConvert {
 
 impl FromArrow for IcebergArrowConvert {}
 
+/// Iceberg sink with `create_table_if_not_exists` option will use this struct to convert the
+/// iceberg data type to arrow data type. Specifically, it will add the field id to the
+/// arrow field metadata, because iceberg-rust and icelake need the field id to be set.
+#[derive(Default)]
```
Comment (xxchan) [inline src/common/src/array/arrow/arrow_iceberg.rs:179]: To further avoid confusion
```suggestion
///
/// Note: this is different from [`IcebergArrowConvert`], which is used to read from/write to
/// an _existing_ iceberg table. In that case, we just need to make sure the data is compatible to the existing schema. 
/// But to _create a new table_, we need to meet more requirements of iceberg.
#[derive(Default)]
```

**Review by xxchan — APPROVED** [review]
LGTM!

**Pull request merged**